numpy/ma/mrecords.py

```diff
@@ -221,7 +221,7 @@ def __getattribute__(self, attr):
             except IndexError:
                 # Couldn't find a mask: use the default (nomask)
                 pass
-            hasmasked = _mask.view((bool, (len(_mask.dtype) or 1))).any()
+            hasmasked = _mask.view((bool, (len(_mask.dtype) or ()))).any()
```

Comment:
I think this needs to be `(len(_mask.dtype),)` else it will run into the deprecation warning here when `len(_mask.dtype) == 1`

Reply:
good catch, fixed